docs/contributing/features.md

```diff
@@ -0,0 +1,24 @@
+# New Features
+
+We review contributions of new features (e.g., new packages, new scripts, new flags) to the core stacks in an attempt to balance the value of the images to the Jupyter community with the cost of maintaining the images over time. Roughly speaking, we think about the following when considering additions:
```

Comment:
IMHO, I'm not sure review is the right verb to use in this sentence because it seems as if the purpose of the review is "to balance the value of the images etc." rather than (what I think you mean) "we accept PR's to balance the value of the images etc."

Reply:
Also, I would consider stronger language than "think" since it's a means of evaluation

Reply:
Put bluntly, every feature requested for inclusion in the images means additional work by a very small number of people to maintain that feature over time. We have to be careful that the features we're agreeing to add are useful enough to justify their inclusion and long-term maintenance here. If we allow everything into these images, they'll become unwieldy to both maintain and use. Personally, I think we're past that point and need to take a more critical eye to what we add directly to this repository while continuing to let the ecosystem grow in other ways (e.g., through lower-maintenance recipes in the documentation, through third-party projects defining additional images).

I think my above commentary is *too* blunt for documentation, however. I don't want to scare off would-be contributors, just make them aware that we'd like to consider an array of options for capturing their features, not just dumping everything into uber-Docker images.

Reply:
This is my suggested rephrasing:

>We review pull requests of new features (e.g., new packages, new scripts, new flags) to balance the value of the images to the Jupyter community with the cost of maintaining the images over time.  Roughly speaking, we evaluate these pull requests based on the following criteria: